File: app/src/main/java/com/streamprobe/android/Stream.kt
package com.streamprobe.android

import androidx.media3.common.MimeTypes

enum class StreamType {
    MP4,
    HLS,
    DASH,
}

data class Stream(
    val title: String,
    val url: String,
    val type: StreamType,
    val mimeType: String? = null,
)

val defaultStreams = listOf(
    Stream(
        title = "Apple Bipbop (VOD)",
        url = "https://devstreaming-cdn.apple.com/videos/streaming/examples/img_bipbop_adv_example_ts/master.m3u8",
        type = StreamType.HLS,
        mimeType = MimeTypes.APPLICATION_M3U8,
    ),
    Stream(
        title = "CBS News (Live)",
        url = "https://news20e7hhcb.airspace-cdn.cbsivideo.com/index.m3u8",
        type = StreamType.HLS,
        mimeType = MimeTypes.APPLICATION_M3U8,
    ),
    Stream(
        title = "Envivio DASH (VOD)",
        url = "https://dash.akamaized.net/envivio/EnvivioDash3/manifest.mpd",
        type = StreamType.DASH,
        mimeType = MimeTypes.APPLICATION_MPD,
    ),
    Stream(
        title = "DASHIF Live — Basic",
        url = "https://livesim.dashif.org/livesim/testpic_2s/Manifest.mpd",
        type = StreamType.DASH,
        mimeType = MimeTypes.APPLICATION_MPD,
    ),
    Stream(
        title = "DASHIF Live — Availability Time Offset (ATO=10s)",
        url = "https://livesim.dashif.org/livesim/ato_10/testpic_2s/Manifest.mpd",
        type = StreamType.DASH,
        mimeType = MimeTypes.APPLICATION_MPD,
    ),
    Stream(
        title = "Unified Streaming — Live + SCTE-35 + ABR",
        url = "https://demo.unified-streaming.com/k8s/live/stable/scte35.isml/.mpd",
        type = StreamType.DASH,
        mimeType = MimeTypes.APPLICATION_MPD,
    ),
)

File: iosApp/iosApp/Streams/Stream.swift
import Foundation

/// A selectable HLS stream in the demo. HLS-only by design (Phase 5 scope):
/// no DASH/MP4/DRM entries. Supports both VOD and live streams. `id` is stable per process so SwiftUI `List` identity holds.
struct Stream: Identifiable, Hashable {
    let id: UUID
    let title: String
    let url: URL
    let isLive: Bool

    init(title: String, urlString: String, isLive: Bool = false) {
        self.id = UUID()
        self.title = title
        // Force-unwrap is acceptable for a hard-coded, reviewed demo catalog.
        self.url = URL(string: urlString)!
        self.isLive = isLive
    }
}

/// Curated public HLS test streams (no DRM). Mirrors the HLS entries from the Android demo
/// plus a couple of widely-used public HLS test assets.
let demoStreams: [Stream] = [
    Stream(
        title: "Apple BipBop — Advanced (TS)",
        urlString: "https://devstreaming-cdn.apple.com/videos/streaming/examples/img_bipbop_adv_example_ts/master.m3u8"
    ),
    Stream(
        title: "Apple BipBop — Advanced (fMP4)",
        urlString: "https://devstreaming-cdn.apple.com/videos/streaming/examples/img_bipbop_adv_example_fmp4/master.m3u8"
    ),
    Stream(
        title: "Apple HEVC — Multi-Subtitle",
        urlString: "https://devstreaming-cdn.apple.com/videos/streaming/examples/bipbop_adv_example_hevc/master.m3u8"
    ),
    Stream(
        title: "Mux — Big Buck Bunny",
        urlString: "https://test-streams.mux.dev/x36xhzz/x36xhzz.m3u8"
    ),
    Stream(
        title: "Mux — Tears of Steel",
        urlString: "https://test-streams.mux.dev/test_001/stream.m3u8"
    ),
    Stream(
        title: "Akamai — Live Test (DVR)",
        urlString: "https://cph-p2p-msl.akamaized.net/hls/live/2000341/test/master.m3u8",
        isLive: true
    ),
    Stream(
        title: "Unified Streaming — Live Demo",
        urlString: "https://demo.unified-streaming.com/k8s/live/stable/live.isml/.m3u8",
        isLive: true
    ),
]
